src/Dfc.CourseDirectory.WebV2/BinaryStorageProvider/BlobStorageBinaryStorageProvider.cs

```diff
@@ -35,11 +36,19 @@ public async Task<bool> TryDownloadFile(string path, Stream destination)
             return true;
         }
 
-        public Task UploadFile(string path, Stream source)
+        public async Task UploadFile(string path, Stream source)
         {
-            var blobRef = _blobClient.GetContainerReference(_containerName).GetBlockBlobReference(path);
+            var container = await GetContainer();
+            var blobRef = container.GetBlockBlobReference(path);
 
-            return blobRef.UploadFromStreamAsync(source);
+            await blobRef.UploadFromStreamAsync(source);
+        }
+
+        private async Task<CloudBlobContainer> GetContainer()
```

Comment:
I believe it's possible to hold the reference to the BlobContainer as a private property (as it's just a client under the hood) to avoid having to execute this codepath an excessive number of times.